### DISCUSSION TITLE
What is the python update policy, going from 3.9.1 to 3.9.2 to ...

### DISCUSSION THREAD

**Discussion by FonsRademakers** [Homebrew/discussions#1109] [Everyday usage]
Hi,

   is the python upgrade from 3.9.1 to 3.9.2 considered a patch level upgrade preserving binary compatibility? I noticed that the python 3.9.1 include directories disappeared when I did a brew update and got python 3.9.2. Is the link to python/3.9/... valid for the whole serie 3.9.*? I have some pre-build binaries build against python 3.9.1 that I ship for brew based systems for which I require python 3.9. Or should I re-build all these binaries with 3.9.2?

Thanks, Fons.

**Comment by Bo98** [comment]
Binary compatibility should work fine for patch level updates.

If linking against brewed Python, ensure you are using the `/usr/local/opt` prefix rather than `/usr/local/Cellar` to avoid links breaking on upgrades.

So:

```
/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9
```

instead of:

```
/usr/local/Cellar/python@3.9/3.9.2_2/Frameworks/Python.framework/Versions/3.9
```

(Formulae do this automatically.)

The general site-packages directory, if that's all you need, is found at:

```
/usr/local/lib/python3.9/site-packages
```

This all assumes the standard Intel Homebrew prefix of `/usr/local`.

**Reply by FonsRademakers** [reply]
Thanks for the clarification. i posted a few days ago a question (https://github.com/Homebrew/discussions/discussions/1071) on the best way to obtain this link without patch level number, is there a way to get this just using the python3 binary?

**Reply by carlocab** [reply]
Why doesn't this work? https://github.com/Homebrew/discussions/discussions/1071#discussioncomment-498611

**Reply by FonsRademakers** [reply]
Sorry guys, I overlooked that you had fixed my example in the meanwhile. Thanks!
